script.js

```diff
@@ -1,4 +1,77 @@
+//user inputs their name
+var startGame = `please enter your name`;
+var winScore = 0;
+var playerTries = 0;
+// user inputs scissors, paper or stone, output will determine win lose or draw or wrong spelling
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+  playerTries = playerTries + 1;
+  //do this if game mode is waiting for enter your name
+  if (startGame == `please enter your name`) {
+    //set the name
+    userName = input;
+    //now that we have the name, switch mode
+    startGame = `SPS game`;
+    playerTries = playerTries - 1;
+    myOutputValue = `Welcome  ` + userName;
+  }
```

Comment:
you could add a return myOutputValue after line 16 to show the output Welcome + userName;